api/Ride/views.py

```diff
@@ -1,18 +1,49 @@
 '''rides views.py'''
-
+from datetime import datetime
 # third party imports
 from flask import make_response, jsonify, request, Blueprint
 from flask.views import MethodView
 # local imports
-from api.models.user import User
+from api.models.ride import Ride
 
 ride_blueprint = Blueprint('ride', __name__)
 
+rides = list()
+
 
-class Ride(MethodView):
+class RideOffer(MethodView):
     ''' a view class with methods to hande ride(s)'''
     def post(self):
         ''' class method which allows user to sign up'''
+        dest = request.json.get('destination')
+        date = request.json.get('date')
+        time = request.json.get('time')
+        meet_p = request.json.get('meetpoint')
+        charges = request.json.get('charges')
+        if dest and date and time and meet_p and charges:
+            try:
+                date = datetime.strptime(date, '%d-%m-%Y').date()
+            except ValueError:
+                return make_response(
+                    jsonify(
+                        {'message':'enter date in the format of dd-mm-YYYY'}
+                        )), 409
+
+            if date < date.today():
+                return make_response(
+                    jsonify({
+                        'message': 'rides can only be of the present and the future'
```

Comment:
Make this message descriptive. You can use short sentence like `ride already expired.`